src/main/java/org/breedinginsight/model/delta/DeltaEntityFactory.java

```diff
@@ -0,0 +1,93 @@
+package org.breedinginsight.model.delta;
+
+import io.micronaut.context.annotation.Bean;
+import io.micronaut.context.annotation.Factory;
+import io.micronaut.context.annotation.Prototype;
+import org.breedinginsight.model.delta.DeltaGermplasm;
+import org.breedinginsight.model.delta.DeltaObservation;
+import org.breedinginsight.model.delta.DeltaObservationUnit;
+import org.breedinginsight.model.delta.DeltaObservationVariable;
+import org.breedinginsight.model.delta.Environment;
+
+import org.apache.commons.lang3.NotImplementedException;
+import org.brapi.v2.model.core.BrAPILocation;
+import org.brapi.v2.model.core.BrAPIStudy;
+import org.brapi.v2.model.core.BrAPITrial;
+import org.brapi.v2.model.germ.BrAPIGermplasm;
+import org.brapi.v2.model.pheno.BrAPIObservation;
+import org.brapi.v2.model.pheno.BrAPIObservationUnit;
+import org.brapi.v2.model.pheno.BrAPIObservationVariable;
+
+@Factory
+public class DeltaEntityFactory {
+
+    public static Experiment makeExperiment(BrAPITrial brAPIObject) {
```

Comment:
The constructor for the factory needs to be implemented with @Inject annotation with a service as a dependency to make it obvious how to extend the entity classes. Include ObservationUnitService#constructPIOFromBrAPIUnit as a dependency.

Reply:
Makes sense. I added `ObservationUnitService` as an injected dependency, I assume that's what you mean.